src/settings.plugin.js

```diff
@@ -248,64 +245,73 @@ wikiaJWPlayerSettingsPlugin.prototype.updateCurrentQuality = function (data) {
 		}
 	}
 };
+// end quality
+
+// captions
+wikiaJWPlayerSettingsPlugin.prototype.onCaptionsChange = function (event) {
+	var emptyCaptionsClass = 'are-captions-empty';
+
+	clearListElement(this.captionsList);
+
+	// tracks always include "off" item
+	if (this.captionsList && event.tracks.length > 1) {
+		var suitableCaptionsTrack = this.getSuitableCaptionsIndex(
+			this.config.selectedCaptions || this.captionLangMap[this.getUserLang()],
+			event.tracks
+		);
+
+		event.tracks.forEach(function (track, index) {
+			var captionItem = document.createElement('li'),
```

Comment:
Can we extract forEach callback as a separate method? This way onCaptionsChange will et a bit smaller and more readable :)